### PR TITLE
Use category's id from filters list when querying

### PR DESCRIPTION
**Pull request by akbarik**
### Please check if the PR fulfills these requirements ###

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
It fixes the bug described in this [issue](https://github.com/deity-io/falcon/issues/426).

**What is the current behavior? (You can also link to an open issue here)**
If you select Men or Women from the Header and then click on any filter in Category list, you will get 0 results, even though in filters it shows that it has products.

### CODE REVIEW

**Comment by Borales** [comment]
In my opinion, we should be dealing with this bug slightly different.
In the suggested PR - we're trying to fix a Magento's bug on the frontend when it cannot apply the provided filters to the query (meaning - fixing a backend-specific issue on the "agnostic" frontend).

I would suggest one of the following solutions:
- exclude `cat` filter in the `magento2-api` package response at all as we have these categories in Men/Women top sub-menus
- we should render `categories` filter block as links, meaning - no filter processing and simply navigate  to the specified sub-category

@cane-lagerwaard @mateuszwozniak @tutok @akbarik what do you guys think?

**Comment by akbarik** [comment]
> * exclude `cat` filter in the `magento2-api` package response at all as we have these categories in Men/Women top sub-menus
No, actually there are more categories that you can recieve using `cat` filter. Try to click on Men/Women -> Tops in the menu. Then expand Category filter group, you'll see several categories there, like Jackets, Tees and so on. They don't exist in the menu.

**Comment by Borales** [comment]
@akbarik sorry for a delayed response.

My point was about fixing this behavior on the `magento2-api` package itself. Apparently, Magento is unable to apply its own filters 🤷‍♂ (selecting any "Category" filter on https://demo.deity.io/men.html shows "0 products"). But if you check a sub-category page (https://demo.deity.io/men/tops-men.html) - now you can select them. I think it has something to do with filtering by category on a "leaf" category (correct me if I'm wrong @sergy-gzh ). 

To summarize: I'd suggest leaving frontend as is, and try to adapt `magento2-api` package to work properly with the filters

**Comment by Borales** [comment]
@akbarik 

> Then expand Category filter group, you'll see several categories there, like Jackets, Tees and so on. They don't exist in the menu.

This is a matter of a frontend template, by default, it renders 2 menu levels only, but the menu could have a lot more levels.

**Comment by akbarik** [comment]
Ok, I see what you mean, @Borales. Agreed, it has to be fixed on magento2-api level. Closing this PR.

**Comment by Borales** [comment]
@akbarik just in case if you'll be working on this issue - https://github.com/deity-io/falcon/issues/426#issuecomment-527918814
